### PR TITLE
JAMES-3926 Migrate Sieve quotas to mutalized quota system

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Needs a rebase though :)

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
```
18:38:57,047 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.3.0:check (check-style) on project james-server-guice-cassandra: You have 2 Checkstyle violations. -> [Help 1]
```

File: server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        testee = new SieveQuotaMigration(usersRepository, oldDAO, newDAO);
+    }
+
+    @Test
+    void shouldMigrateGlobalLimit() throws Exception {
+        oldDAO.setQuota(QuotaSizeLimit.size(32)).block();
+
+        testee.apply();
+
+        assertThat(oldDAO.getQuota().block())
```
Comment (Arsnael) [inline server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java:94]: newDao?

File: server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            .contains(QuotaSizeLimit.size(32));
+    }
+
+    @Test
+    void shouldNotFailWhenNoGlobalLimit() throws Exception {
+        testee.apply();
+
+        assertThat(oldDAO.getQuota().block())
```
Comment (Arsnael) [inline server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java:102]: newDao?

File: server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            .isEmpty();
+    }
+
+    @Test
+    void shouldNotFailWhenNoUserLimit() throws Exception {
+        testee.apply();
+
+        assertThat(oldDAO.getQuota(BOB).block())
```
Comment (Arsnael) [inline server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java:110]: newDao?

File: server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            .isEmpty();
+    }
+
+    @Test
+    void shouldNotFailWhenNoSpaceUsed() throws Exception {
+        testee.apply();
+
+        assertThat(oldDAO.spaceUsedBy(BOB).block())
```
Comment (Arsnael) [inline server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/migration/SieveQuotaMigrationTest.java:118]: newDao?

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

File: server/apps/cassandra-app/src/main/java/org/apache/james/CassandraJamesServerConfiguration.java [hunk 5]
```diff
@@ -142,6 +149,14 @@ public CassandraJamesServerConfiguration build() {
                 }
             });
 
+            boolean quotaCompatibilityMode = this.quotaCompatibilityMode.orElseGet(() -> {
+                try {
+                    return configurationProvider.getConfiguration("cassandra").getBoolean("quota.compatibility.mode", false);
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/apps/cassandra-app/src/main/java/org/apache/james/CassandraJamesServerConfiguration.java:154]: Was intentionnal as this is used only for migration purposes.

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1700/12/testReport/junit/org.apache.james.webadmin.integration.rabbitmq/RabbitMQWebAdminServerIntegrationTest/getCurrentVersionShouldReturnNullForCurrentVersionAsBeginning/

-> related

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1700/15/

related?

**Pull request merged**